content/articles/form-validation-in-vue.js-using-veevalidate/index.md

```diff
@@ -0,0 +1,469 @@
+
+### Prerequisites
+
+To follow along in this article, it is helpful to have the following:
+
+- [Node](https://nodejs.org) installed
+- [Vue](https://vuejs.org/) installed
+- [yarn](https://classic.yarnpkg.com/en/docs/install)
+- Basic knowledge of JavaScript programming.
+- Basic knowledge and practice of Vue.
+- Text editor installed. Preferably [VS Code](https://code.visualstudio.com/)
+
+### Overview
+
+- [Why Client Side Validation](#why-client-side-validation)
+- [Project SetUp](#project-setup)
+- [Introducing VeeValidate](#introducing-veevalidate)
+- [Form validation](#form-validation)
+- [Custom Error Messages](#custom-error-messages)
+- [Validation Triggers](#validation-triggers)
+- [Add Alert Message](#add-alert-message)
+
+### Why Client-Side Validation
+
+Client-side validation is essential, it provides users with immediate feedback thus increasing user experience and saves on application resources by eliminating the response request cycle process hence saving on bandwidth and time. This is not to say that server-side validation should not be performed on an application, client-side validation can be disabled in the browser, it is wise to perform validation both on the client and the server.
+
+### Project SetUp
+
+Make sure you have Vue configured in your machine. Create a new Vue app with the above command:
+
+```bash
+vue create vue-form-validation
+```
+
+We are going to start by installing tailwind and setting up a demo template for practice. Install [tailwind plugin][https://www.npmjs.com/package/vue-cli-plugin-tailwind]
+
+In the newly created repository, install the plugin using the above command:
+
+```bash
+vue add tailwind
+```
+
+Let's clean up the default Vue setup by deleting the HelloWorld component and its dependencies. Create a new component and set up the tailwind template, copy and paste the template(Tailwind Form Template) body elements from the above [pen][https://codepen.io/dev_njunu]. Fire up the server and navigate to the local port, you will have a registration form. Feel free to customize it to your liking.
+
+### Introducing VeeValidate
+
+[VeeValidate][https://vee-validate.logaretm.com/v4/] is an effective validation library it is tested to work with most browsers. Before we get into working with the library let's explore it. The above library provides us with two ways of form validation, components and composition API. We are going to adopt the components approach in the above tutorial.
+
+We are provided with the above components:
+
+- **Form** component, this renders a form, wraps all form elements.
+- **Field** component, represents form inputs and renders any HTML element.
+- **ErrorMessage** component, displays an error message for a field.
+
+Let's define the validation process and see how it relates to the mentioned components:
+
+1. Wrap your form with the **Form** component, overwriting the default _form_ elements.
```

Comment:
Fix the numbering issue here.

Reply:
hi, thanks for the comment but think the md is ok(https://guides.github.com/features/mastering-markdown/)

Reply:
Ah thanks. Then that makes sense. Thanks, I did not know about that!